deploy/examples/operator-with-tracing.yaml

```diff
@@ -0,0 +1,52 @@
+# this is an example `operator.yaml` with all the bits required to get the Jaeger Operator
+# to generate spans for its own operations. See the operator documentation for more information about this feature.
+apiVersion: apps/v1
+kind: Deployment
+metadata:
+  name: jaeger-operator
+  namespace: observability
+spec:
+  replicas: 1
+  selector:
+    matchLabels:
+      name: jaeger-operator
+  template:
+    metadata:
+      labels:
+        name: jaeger-operator
+    spec:
+      serviceAccountName: jaeger-operator
+      containers:
+      - name: jaeger-operator
+        image: jaegertracing/jaeger-operator:1.16.0 # operator tracing is available since 1.16.0
+        ports:
+        - containerPort: 8383
+          name: metrics
+        args: ["start", "--tracing-enabled=true"]
+        imagePullPolicy: Always
+        env:
+        - name: WATCH_NAMESPACE
+          value: ""
+        - name: POD_NAME
+          valueFrom:
+            fieldRef:
+              fieldPath: metadata.name
+        - name: POD_NAMESPACE
+          valueFrom:
+            fieldRef:
+              fieldPath: metadata.namespace
+        - name: OPERATOR_NAME
+          value: "jaeger-operator"
+      - name: jaeger-agent
+        image: jaegertracing/jaeger-agent:1.15.1
```

Comment:
This should also be `1.16.0` - although not ideal hard coding the tag. Would it be better to use `latest` and have comment that it is only implemented in `1.16.0`?

Reply:
True. I thought about using 1.15.1 to point out that the current latest version of the Agent can be used, but it's better if both versions are aligned.